crates/oxc_codegen/tests/integration/unit.rs

```diff
@@ -63,6 +63,33 @@ fn unicode_escape() {
     test("console.log('🧑‍🤝‍🧑');", "console.log(\"🧑‍🤝‍🧑\");\n");
 }
 
+#[test]
+fn regex() {
```

Comment:
Added some tests here, It would've caught the issue we had.
I'm not sure exactly how to write the oxlint integration test, Maybe we can write a temp file and apply a bunch of fixes to it to make sure the whole lint cycle is working as expected but I feel that should be its own PR.

Reply:
There are also tests available for rules with fixers, The only reason that it was uncaught was that none of our regex fixers were using codegen, The root of the issue was that codegen was panicking when the source wasn't available to it.
These tests make sure we generate the correct regexes with or without the source. Now having the original source in the linter becomes a decision rather than a bug and that's why I have difficulties imagining a useful unit test for it.